pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -352,3 +377,71 @@ func isMaster(machine mapiv1.Machine, client client.Client) bool {
 	}
 	return false
 }
+
+// isMachineDisruptionAllowed returns true if the provided MachineDisruptionBudget allows any disruption
+func isMachineDisruptionAllowed(mdb *healthcheckingv1alpha1.MachineDisruptionBudget, maxDisruptedMachinSize int) bool {
```

Comment:
I think `isMachineDisruptionAllowed` better describes the purpose of this method, because it checks if remediation allowed only in the context of the machine disruption budget, it possible that in future we will have additional conditions when we want to prevent remediation and in this case, we can introduce `isMachineRemediationAllowed` and call to `isMachineDisruptionAllowed` as part of it.